pkg/distributors/contracts/MultiRewards.sol

```diff
@@ -70,191 +75,294 @@ contract MultiRewards is IMultiRewards, IDistributor, ReentrancyGuard, Temporari
+            rewardData[pool][rewarder][rewardsToken].rewardPerTokenStored.add(
+                lastTimeRewardApplicable(pool, rewarder, rewardsToken)
+                    .sub(rewardData[pool][rewarder][rewardsToken].lastUpdateTime)
+                    .mulDown(rewardData[pool][rewarder][rewardsToken].rewardRate)
+                    .divDown(_totalSupply[pool])
             );
     }
 
     /**
      * @notice Calculates the amount of `rewardsToken` that `account` is able to claim
+     * from a particular rewarder
      */
     function earned(
-        IERC20 stakingToken,
+        IERC20 pool,
+        address rewarder,
         address account,
         IERC20 rewardsToken
     ) public view returns (uint256) {
         return
-            _balances[stakingToken][account]
+            _balances[pool][account]
                 .mulDown(
-                rewardPerToken(stakingToken, rewardsToken).sub(
-                    userRewardPerTokenPaid[stakingToken][account][rewardsToken]
+                rewardPerToken(pool, rewarder, rewardsToken).sub(
+                    userRewardPerTokenPaid[pool][rewarder][account][rewardsToken]
                 )
             )
-                .divDown(1e18)
-                .add(rewards[stakingToken][account][rewardsToken]);
+                .add(rewards[pool][rewarder][account][rewardsToken]);
+    }
+
+    /**
+     * @notice Calculates the total amount of `rewardsToken` that `account` is able to claim
+     */
+    function totalEarned(
+        IERC20 pool,
+        address account,
+        IERC20 rewardsToken
+    ) public view returns (uint256 total) {
+        for (uint256 r; r < _rewarders[pool][rewardsToken].length(); r++) {
+            total = total.add(earned(pool, _rewarders[pool][rewardsToken].at(r), account, rewardsToken));
+        }
     }
 
-    function getRewardForDuration(IERC20 stakingToken, IERC20 rewardsToken) external view returns (uint256) {
+    function getRewardForDuration(
+        IERC20 pool,
+        address rewarder,
+        IERC20 rewardsToken
+    ) external view returns (uint256) {
         return
-            rewardData[stakingToken][rewardsToken].rewardRate.mulDown(
-                rewardData[stakingToken][rewardsToken].rewardsDuration
+            rewardData[pool][rewarder][rewardsToken].rewardRate.mulDown(
+                rewardData[pool][rewarder][rewardsToken].rewardsDuration
             );
     }
 
     /* ========== MUTATIVE FUNCTIONS ========== */
-
-    function setRewardsDistributor(
-        IERC20 stakingToken,
-        IERC20 rewardsToken,
-        address rewardsDistributor
-    ) external onlyOwner {
-        rewardData[stakingToken][rewardsToken].rewardsDistributor = rewardsDistributor;
-    }
-
-    function stake(IERC20 stakingToken, uint256 amount) external {
-        stake(stakingToken, amount, msg.sender);
+    function stake(IERC20 pool, uint256 amount) external {
+        stake(pool, amount, msg.sender);
```

Comment:
I'd consider removing this version and just leaving the 3 argument one - also, external overloads are a pain in client code.

Reply:
(that is, assuming we care about arbitrary recipients)

Reply:
Will cover this in #659